### PR TITLE
Add GitHub Actions workflow to publish files to production

### LINKED ISSUE #1717: Potential New Way of Moving Documents to Production Website
## State Of Things Today

We have to move documents between git branches. The `master` branch is for documents in development, and the `prod` branch is for documents that should be displayed on our websites (specifically www.mozilla.org). This is a fine system for people very familiar with git, but for those who don't normally use git for their day-to-day work (i.e. most of the folks working with these documents) it can get confusing very quickly. Therefore we're in a state where the editors of these documents can update them themselves, but can't publish them without help from myself or others on my team.

## Proposal

I am thinking it might be easier if we keep both documents in development and for production in the default branch (`master`), but in separate directories. This would make "publishing" a document as easy as dragging it into the right place in the directory structure and commiting and pushing that change.

I'm curious if this sounds like it would be a superior solution and whether it would really enable full self-service. 

I'm thinking the directory structure could be something like what follows:

```
/
    /in-progress
        /de
           firefox-privacy.md
        /en
            firefox-privacy.md
    /published
        /de
           firefox-privacy.md
        /en
            firefox-privacy.md
```

@flodolo @peiying2

### PR DESCRIPTION
**Pull request by pmac**
This will add a workflow action that will take a comma separated list of files as input (or "ALL" by default if all files are desired) and create a pull-request for updating the content of the files and all of the translations thereof. After reviewing the PR it can be merged to publish all of the file changes to the production website.

Fix #1717

### CODE REVIEW

**Comment by pmac** [comment]
If this works I'll update the README to reflect the updated production push procedure.

**Review by flodolo — CHANGES_REQUESTED** [review]
A few fixes needed, but this seems to work great.

**Comment by pmac** [comment]
Thanks for the review! I'll fix it up.

File: .github/scripts/copy_files.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+def copy_files(filename, dest):
+    dest_path = Path(dest)
+    src_path = Path(".")
+    if filename == "ALL":
+        file_paths = src_path.glob(f"*/*.md")
+    else:
+        file_paths = src_path.glob(f"*/{filename}")
+
+    for fp in file_paths:
```
[the remark this answers is not in the document]
Comment (pmac) [reply .github/scripts/copy_files.py:19]: Good idea. Should probably exit with a non-zero exit code as well.

File: .github/workflows/prod_push.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        run : |
+          git config --global user.email 'MozmarRobot@users.noreply.github.com'
+          git config --global user.name 'MozMEAO Bot'
+      - name: Update files in repository
+        run: |
+          cd clone
+          git fetch origin '+refs/heads/*:refs/heads/*' --update-head-ok
+
+          # Use temporary folder to store files from the default branch
+          TMP_DIR=$(mktemp)
```
[the remark this answers is not in the document]
Comment (pmac) [reply .github/workflows/prod_push.yml:43]: Ah. I was getting too fancy :) Thanks.

**Comment by pmac** [comment]
Updates are done. Thanks again for taking a look and for the excellent suggestion flod! After this merges I'll file a PR to update the README instructions and send an email about it.

**Review by flodolo — APPROVED** [review]
Did some [more testing](https://github.com/flodolo/legal-docs/actions/workflows/prod_push.yml) on my fork, and it works great!

For documentation: one of the side effects of the approach you used for the search, is that you can use wildcards in the input field (e.g. `mdn*`). Not sure if that was planned, but it's very handy.

**Comment by pmac** [comment]
> For documentation: one of the side effects of the approach you used for the search is that you can use wildcards in the input field (e.g. `mdn*`). Not sure if that was planned, but it's very handy.

It wasn't exactly intended, but it's a good point. I'll include that in the documentation.

**Pull request merged**